sys/net/gnrc/netreg/gnrc_netreg.c

```diff
@@ -177,7 +177,10 @@ void gnrc_netreg_unregister(gnrc_nettype_t type, gnrc_netreg_entry_t *entry)
     if (entry->type == GNRC_NETREG_TYPE_MBOX) {
         msg_t msg;
         while (mbox_try_get(entry->target.mbox, &msg)) {
-            gnrc_pktbuf_release_error(msg.content.ptr, EBADF);
+            if ((msg.type == GNRC_NETAPI_MSG_TYPE_RCV)
+                    || (msg.type == GNRC_NETAPI_MSG_TYPE_SND)) {
+                gnrc_pktbuf_release_error(msg.content.ptr, EBADF);
+            }
         }
     }
```

Comment:
Again: I think it is cleaner, to move this whole block to (the yet to be created) `gnrc_sock_close()` (which also would call `gnrc_netreg_unregister()`. From our offline discussion yesterday, I think the content of the `mbox` is much more use case dependent than previously thought and only `gnrc_sock` really knows what it all put into it.

Reply:
But `GNRC_NETAPI_MSG_TYPE_SND/RCV` messages aren't exclusive to `gnrc_sock`, or am I missing something?

Reply:
No, but `_TIMEOUT_MSG_TYPE` and whatever `gnrc_sock` might throw in here (potentially with packet context) in here, is.

Reply:
Alterntaively `SND`/`RCV` could be cleaned out here, but there should also be a centralized clean-out within `gnrc_sock` to properly handle the rest.